src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -246,9 +255,76 @@ public void initialize() {
             * Constants.Lift.ELEVATOR_MOTOR_ENCODER_DIFFERENCES_SCALAR_INCHES_PER_DEGREE);
   }
 
+  /** Sets the desired position, which the lift may not go to directly. */
+  public void setDesiredPosition(LiftPosition pos) {
+    desiredPosition = pos;
+  }
+
+  /** True if the lift is at the queried position. */
+  private boolean atPosition(LiftPosition positionToCheck) {
+    double armThresholdDegrees =
+        positionToCheck == LiftPosition.STARTING
+            ? Cal.Lift.ARM_START_THRESHOLD_DEGREES
+            : Cal.Lift.ARM_THRESHOLD_DEGREES;
+    double elevatorThresholdInches =
+        positionToCheck == LiftPosition.STARTING
+            ? Cal.Lift.ELEVATOR_START_THRESHOLD_INCHES
+            : Cal.Lift.ELEVATOR_THRESHOLD_INCHES;
+    double elevatorPositionToCheckInches = liftPositionMap.get(positionToCheck).getFirst();
+    double armPositionToCheckDegrees = liftPositionMap.get(positionToCheck).getSecond();
+    double elevatorPositionInches = elevatorEncoder.getPosition();
+    double armPositionDegrees = armEncoder.getPosition();
+    if (Math.abs(armPositionDegrees - armPositionToCheckDegrees) > armThresholdDegrees) {
+      return false;
+    }
+
+    if (Math.abs(elevatorPositionInches - elevatorPositionToCheckInches)
+        > elevatorThresholdInches) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /** Converts a lift position to a relative lift position (above or below starting position). */
+  private static LiftPositionStartRelative getRelativeLiftPosition(LiftPosition pos) {
+    switch (pos) {
+      case STARTING:
+        return LiftPositionStartRelative.AT_START;
+      case SCORE_MID:
+      case SCORE_HIGH:
```

Comment:
Why are these empty?

Reply:
It's a "fallthrough." If you don't break, the switch continues on to the code in the next case. Sometimes it's considered an anti-pattern but here I think it's okay.

Reply:
Ok. I just read an article about it and I think I understand it now.